### PR TITLE
x:Bind functions support

### LINKED ISSUE #347: Add support for x:Bind Functions
As documented at https://docs.microsoft.com/en-us/windows/uwp/data-binding/function-bindings
Functions in x:Bind would be a real boon because it is the only way in UWP to access static Properties or Functions. 

Example:

                <TextBlock Text="{x:Bind UI:Translations.TestMe}"/>

### LINKED ISSUE #934: x:Bind to fields doesn't work
## I'm submitting a...

Feature request

## Current behavior

Can't bind to fields with `x:Bind`.

## Expected behavior

Can.

## Minimal reproduction of the problem with instructions

XAML:
```xaml
<TextBlock Text="{x:Bind PrivateField}"/>
<TextBlock Text="{x:Bind PublicField}"/>
```

Code-behind:
```csharp
		public string PublicField = "Enemy";
		private string PrivateField = "Ryan";
```

## Environment

Nuget Package:

Package Version(s):

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

### LINKED ISSUE #2260: Bindings property when using x:Bind
## What would you like to be added:
A Bindings property of a usercontrol, generated by the compiler, when adding x:Bind bindings.
## Why is this needed:
This is needed to be able to access the x:Bind bindings defined in XAML. When a datacontext changes, the bindings need to be updated. In UWP this is fixed easily by calling Bindings.Update() in the DataContextChanged eventhandler in the code behind.
## For which Platform:

- [ ] iOS
- [ ] Android
- [ x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?
I am just developing for UWP and WebAssembly, but maybe this applies also for other platforms other than UWP and WebAssembly

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes #2429, fixes #347, fixes #2260, fixes #934

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Adds support for x:Bind functions: https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-bind-markup-extension

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlConstants.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ internal static class XamlConstants
 		public const string BaseXamlNamespace = RootUINamespace + ".Xaml";
 		public const string UnoXamlNamespace = "Windows.UI.Xaml";
 
+		public const char XBindSubstitute = '↔';
```
Comment (carldebilly) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlConstants.cs]: What is that??

Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlConstants.cs]: See:
https://github.com/unoplatform/uno/pull/2429/files#diff-8791012553acc33c896cb7e78896157bR134

File: doc/articles/uno-development/Uno-UI-xBind-architecture.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# x:Bind in Uno Platform
+
+
+
+## Overview of the x:Bind feature in WinUI
```
Comment (davidjohnoliver) [inline doc/articles/uno-development/Uno-UI-xBind-architecture.md]: Love the detailed findings and implementation discussion 👍

**Comment by jeromelaban** [comment]
> Amazing to see everything except BindBack got included 👍

Almost! There's also the `Bindings.Update()` that is not completely doing its job. Will update in a later PR.

**Pull request merged**